packages/lesmis/src/utils/makeEntityLink.js

```diff
@@ -5,8 +5,9 @@
  */
 import { generatePath } from 'react-router-dom';
 
-export const makeEntityLink = (entityCode, view = 'dashboard') =>
-  generatePath('/:entityCode/:view', {
+export const makeEntityLink = (locale, entityCode, view = 'dashboard') =>
+  generatePath(`/:locale/:entityCode/:view`, {
```

Comment:
Is it possible to use `useUrlParams` within this function, and get the `locale` there, rather than requiring it to be passed through from every place that calls `makeEntityLink`? I'm guessing not or you would have done it! Probably that would require that `makeEntityLink` itself would need to be treated like a hook, and only called outside of loops and conditionals etc.

Reply:
Oh yeah well that does make things bloody easy... let's just do that! I have a sneaking feeling we might come to regret it if react-router and window.location somehow come out of sync (even for a moment in the component lifecycle), but it's so easy we should just do it and fix it if that happens.